pkg/controller/ibmblockcsi/ibmblockcsi_controller.go

```diff
@@ -263,15 +259,69 @@ func (r *ReconcileIBMBlockCSI) updateStatus(instance *ibmblockcsi.IBMBlockCSI) e
 	if instance.Status.ControllerReady && instance.Status.NodeReady {
 		phase = csiv1.DriverPhaseRunning
 	} else {
+		if originalStatus.ControllerReady && !instance.Status.ControllerReady {
+			controllerRestart = true
+		}
+		if originalStatus.NodeReady && !instance.Status.NodeReady {
+			nodeRolloutRestart = true
+		}
 		phase = csiv1.DriverPhaseCreating
 	}
 	instance.Status.Phase = phase
 	instance.Status.Version = oversion.DriverVersion
 
-	// no need to push to status to API Server here.
+	if !reflect.DeepEqual(originalStatus, instance.Status) {
+		reqLogger.Info("updating IBMBlockCSI status", "name", instance.Name, "from", originalStatus, "to", instance.Status)
+		sErr := r.client.Status().Update(context.TODO(), instance.Unwrap())
+		if sErr != nil {
+			return sErr
+		}
+	}
+
+	if controllerRestart {
+		reqLogger.Info("restarting controller")
```

Comment:
"due to status change"? I think more info is needed.. if all we see in logger is restarting it can be confusing (especially for users)

Reply:
in node too

Reply:
done